src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -778,11 +767,27 @@ protected TNode AddError<TNode>(TNode nodeOrToken, ErrorCode code, params object
                 Debug.Assert(nodeOrToken.Width > 0 || nodeOrToken.RawKind is (int)SyntaxKind.EndOfFileToken);
                 return WithAdditionalDiagnostics(nodeOrToken, MakeError(nodeOrToken, code, args));
             }
+            else
+            {
+                var (offset, width) = this.GetDiagnosticSpanForMissingNodeOrToken(nodeOrToken);
+                return WithAdditionalDiagnostics(nodeOrToken, MakeError(offset, width, code, args));
```

Comment:
core change here.  instead of hte logic for this computation being here (and thus, only performed if AddError itself was calleD).  It's now moved into GetDiagnosticSpanForMissingNodeOrToken, so that anyone that is computing diagnostics for missing thigns gets the same logic.